_specifications/base/0.9/specification.md

```diff
@@ -0,0 +1,904 @@
+---
+title: Base Protocol Specification
+shortTitle: 0.9
+layout: specifications
+sectionid: base-0-9
+toc: base-0-9-toc
+fullTitle: Base Protocol Specification - 0.9
+index: 3
+toolsVersion: 0.9
+---
+
+## <a href="#baseProtocol" name="baseProtocol" class="anchor"> Base Protocol </a>
+
+The purpose of the base protocol is to create an abstraction of common editor extensibility patterns into its own specification, independently of the presence of a language server powering such extension. In particular, concepts such as server and client capabilities, the initialization and shutdown request, and the structure of requests and notifications, were originally part of the language server protocol, but there is nothing about them that should be exclusive to language service extensions.
+
+A motivating example is the [Build Server Protocol](https://build-server-protocol.github.io/). As its specification describes, all notifications and requests are defined using [the base definitions](#basicJsonStructures) of LSP, and messages such `InitializeBuild`, `OnBuildInitialized`, and `OnBuildExit` all match almost exactly with their LSP counterparts. By implementing the base protocol, a server can support multiple protocol specifications without having to reimplement the "common boilerplate" that all of them share.
+
+### <a href="#headerPart" name="headerPart" class="anchor"> Header Part </a>
```

Comment:
I like the motivation section. One clarification we should add is the following: the base protocol is experimental / preview and depending on the feedback we receive it might not be removed again.

I want to make sure that we don't have to maintain it in cases it is not adopted or used by other specific protocols.